pkg/filemanager/filemanager.go

```diff
@@ -44,13 +46,91 @@ func NewFileManager(path string) (FileManager, error) {
 		return NewGCSStorageFile(path)
 	case strings.Contains(path, "blob.core.windows.net"):
 		return NewAzureBlobFile(path)
+	case strings.HasPrefix(path, "http://"), strings.HasPrefix(path, "https://"):
+		return NewCustomS3File(path)
 	case path == "":
 		return nil, errors.New("empty path")
 	default:
 		return NewSystemFile(path), nil
 	}
 }
 
+type CustomS3File struct {
+	s3Client *s3.Client
+	bucket   string
+	key      string
+}
+
+func NewCustomS3File(path string) (*CustomS3File, error) {
+	u, err := url.Parse(path)
+	if err != nil {
+		return nil, err
+	}
+
+	clPath := gp.Clean(u.Path)
+
+	if len(strings.Split(clPath, "/")) < 3 {
+		return nil, fmt.Errorf("invalid s3 path: %s", path)
+	}
+
+	// Extract bucket and path from url
+	bucket, key, _ := strings.Cut(strings.TrimLeft(clPath, "/"), "/")
+
+	if bucket == "" || key == "" {
+		return nil, fmt.Errorf("invalid s3 path: %s", path)
+	}
+
+	const defaultRegion = "us-east-1"
```

Comment:
Do you need to hardcode the region?
I suspect this won't be correct for the majority of the people. If it's a required parameter I would rather fail with an error early and let the user fix it.

Reply:
For the region I used this: https://github.com/minio/minio/discussions/15063 which is also described in the Minio documentation but both are based on using the AWS cli and not the SDK. I realize that's quite a bold assumption. After reading this: https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/endpoints/ it's probably better to make it customizable as there is already an environment variable present for that `AWS_REGION`.

Reply:
After reviewing the AWS SDK docs for endpoint configuration I changed the implementation which made it a little bit more straightforward. See also: https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/endpoints/